src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleSerializer.cs

```diff
@@ -0,0 +1,100 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.Health.Fhir.Core.Extensions;
+using Microsoft.Health.Fhir.Shared.Core.Features.Search;
+
+namespace Microsoft.Health.Fhir.Api.Features.Resources.Bundle
+{
+    public static class BundleSerializer
+    {
+        private static readonly JsonWriterOptions WriterOptions = new JsonWriterOptions { Encoder = System.Text.Encodings.Web.JavaScriptEncoder.UnsafeRelaxedJsonEscaping };
+
+        public static async Task Serialize(Hl7.Fhir.Model.Bundle bundle, Stream outputStream)
+        {
+            using (Utf8JsonWriter writer = new Utf8JsonWriter(outputStream, WriterOptions))
+            {
+                writer.WriteStartObject();
+
+                writer.WriteString("resourceType", bundle.ResourceType.ToString());
+                writer.WriteString("id", bundle.Id);
+
+                SerializeMetadata();
+
+                writer.WriteString("type", bundle.Type?.ToCamelCaseString());
```

Comment:
All the provided types have the FHIR value in EnumLiteralAttribute. It's not always true that the FHIR value will be camelcase.

Reply:
Thanks, switched to using the GetLiteral() extension method